pkg/server/smb/server.go

```diff
@@ -25,6 +25,20 @@ func normalizeWindowsPath(path string) string {
 	return normalizedPath
 }
 
+func normalizeMappingPath(path string) string {
+	items := strings.Split(path, "\\")
+	parts := []string{}
+	for _, s := range items {
+		if len(s) > 0 {
+			parts = append(parts, s)
+			if len(parts) == 2 {
```

Comment:
is `normalizeMappingPath` necessary here? I remember csi-proxy has already convert `/` to `\\` in somewhere.
this func assumes path only has two `\\`, it's not always that case.

Reply:
`normalizeMappingPath` gets from a full smb path the smb share name that we will use for mapping.
For example:
```
\\hostname\share\a\b\c\d  => \\hostname\share
\\hostname\share => \\hostname\share
```

csi-driver-smb has the same function for reference counting. Unfortunately, we can't send normalized value to csi-proxy because after create smb global mapping it also makes link to full remote path. I tried to avoid change of contract between csi-driver-smb and csi-proxy. So, this function exists in both repos.

Reply:
while if you change `\\hostname\share\a\b\c\d` to `\\hostname`, that already changes the mounting source, e.g. if user wants to access file under `\\hostname\share\a\b\c\d`, it's in the root dir, while with this PR, it's under `a\b\c\d` after mount.

Reply:
No.
When user want to mount `\\hostname\share\subpath` now we call:
```
New-SmbGlobalMapping -RemotePath "\\hostname\share\subpathA" -Credential $creds
New-Item -ItemType SymbolicLink "...\globalmount" -Target "\\hostname\share\subpathA"
```
This is a problem because we can't do:
```
New-SmbGlobalMapping -RemotePath "\\hostname\share\subpathB" -Credential $creds
New-Item -ItemType SymbolicLink "...\globalmount" -Target "\\hostname\share\subpathB"
```

This PR do:
```
New-SmbGlobalMapping -RemotePath "\\hostname\share" -Credential $creds
New-Item -ItemType SymbolicLink "...\globalmount" -Target "\\hostname\share\subpathA"
```
Then
```
New-Item -ItemType SymbolicLink "...\globalmount" -Target "\\hostname\share\subpathB"
```
The last mount doesn't call `New-SmbGlobalMapping` beause it already mounted.
And it works.

Reply:
So, user will have result as expected.

Reply:
subpath might be empty

Reply:
that makes sense, thanks for the explanation.